File: demo-app/app/src/javaApp/java/com/launchkey/android/authenticator/demo/util/KeyStorage.java
package com.launchkey.android.authenticator.demo.util;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class KeyStorage {

    private static final String PREF_KEY = "auth-sdk-key";

    private static volatile KeyStorage sInstance;

    public static KeyStorage getInstance(final Context context) {

        if (sInstance == null) {
            synchronized (KeyStorage.class) {
                if (sInstance == null) {
                    sInstance = new KeyStorage(context.getApplicationContext());
                }
            }
        }

        return sInstance;
    }

    private SharedPreferences mPreferences;

    private KeyStorage(final Context context) {

        mPreferences = PreferenceManager.getDefaultSharedPreferences(context.getApplicationContext());
        load();
    }

    private String mKey;

    private void load() {

        mKey = mPreferences.getString(PREF_KEY, null);
    }

    private void save() {

        mPreferences.edit().putString(PREF_KEY, mKey).apply();
    }

    public boolean hasKey() {

        return getKey() != null;
    }

    public String getKey() {
        return mKey;
    }

    public void clearKey() {

        setKey(null);
    }

    public void setKey(String key) {
        mKey = key;
        save();
    }
}

File: demo-app/app/src/kotlinApp/java/com/launchkey/android/authenticator/demo/util/KeyStorage.kt
package com.launchkey.android.authenticator.demo.util

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager

class KeyStorage private constructor(context: Context) {

    private val mPreferences: SharedPreferences

    private var mKey: String? = null

    var key: String?
        get() = mKey
        set(key) {
            mKey = key
            save()
        }

    init {

        mPreferences = PreferenceManager.getDefaultSharedPreferences(context.applicationContext)
        load()
    }

    private fun load() {

        mKey = mPreferences.getString(PREF_KEY, null)
    }

    private fun save() {

        mPreferences.edit().putString(PREF_KEY, mKey).apply()
    }

    fun hasKey(): Boolean {

        return key != null
    }

    fun clearKey() {

        key = null
    }

    companion object {

        private val PREF_KEY = "auth-sdk-key"

        @Volatile
        private var sInstance: KeyStorage? = null

        fun getInstance(context: Context): KeyStorage? {

            if (sInstance == null) {
                synchronized(KeyStorage::class.java) {
                    if (sInstance == null) {
                        sInstance = KeyStorage(context.applicationContext)
                    }
                }
            }

            return sInstance
        }
    }
}